docs/sources/developers/plugins/migration-guide.md

```diff
@@ -4,77 +4,451 @@ title = "Plugin migration guide"
+}
+
+func (d *SampleDatasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+  log.DefaultLogger.Info("QueryData called", "request", req)
+  // logic for querying your datasource.
+}
+
+func (d *SampleDatasource) CheckHealth(_ context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
+  log.DefaultLogger.Info("CheckHealth called", "request", req)
+  // The main use case for these health checks is the test button on the
+  // datasource configuration page which allows users to verify that
+  // a datasource is working as expected.
+}
+```
+
+### Unsigned backend plugins will not be loaded
+
+We strongly recommend our Grafana users not to allow unsigned plugins in their Grafana installation. By allowing unsigned plugins, we can’t guarantee the authenticity of the plugin which could compromise the security of your Grafana installation.
+
+This means that you, as a plugin developer, need to get your plugin signed if you want it to be able to run on all Grafana installations.
+
+Follow the following [steps](https://grafana.com/docs/grafana/latest/developers/plugins/sign-a-plugin/#sign-a-plugin) to get instructions on how to set up plugin signing.
+
+You will still be able to run and develop your plugin during development by running your Grafana [instance in development mode](https://grafana.com/docs/grafana/latest/administration/configuration/#app_mode).
+
+### Time series data can now be structured in a wide or many format.
+
+Starting in Grafana 8 time series data can either be structured in either `wide` or `many` format. Prior to this version time series data was delivered in the following format (which is the `many` format).
+
+| time                 | temperature |
+| -------------------- | ----------- |
+| 2021-05-08T07:31:45Z | 23          |
+| 2021-05-08T09:31:45Z | 25          |
+| 2021-05-08T11:31:45Z | 27          |
 
-It's written for:
+| time                 | humidity |
+| -------------------- | -------- |
+| 2021-05-08T07:31:45Z | 60       |
+| 2021-05-08T09:31:45Z | 55       |
+| 2021-05-08T11:31:45Z | 55       |
 
-- Plugin authors who want to migrate their plugins to Grafana 7.0+.
-- Plugin users who are using custom plugins and want to know whether they can upgrade to Grafana 7.0 without losing functionality.
+> Data delivered as an array of data frames (time field needs to be repeated).
 
-> If you've successfully migrated your plugin from Angular to React, please [submit an issue on GitHub](https://github.com/grafana/grafana/issues/new?title=Docs%20feedback:%20/developers/plugins/migration-guide.md) and share your experiences with us so that we can improve this guide!
+That made it possible to detect time series data by inspecting the data frame. If it had two fields (time + value) it most likley was time series data. This is no longer possible since the same data can be delivered in the following `wide` format.
```

Comment:
Can you clarify what the new detection method should be? This will help the user understand what to do instead.

Reply:
So there are basically no new detection method. They need to support both formats.

Reply:
I'm confused by the last sentence. Is it saying that _Grafana_ can no longer detect time series data because it has more support for wide format?

I like the context provided by this, explaining _why_ things are the way they are, but I would reword this to focus very explicitly on _Grafana_ behaviour (old and new) and expectations of the plugin developer.

Reply:
I think the main reason behind moving in this direction was that with the `wide` time-series representation multiple data frames can be "merged" in case the values have the same timestamps, and thus the size of the representation will be smaller.

While using the `many` format (one value per timestamp) it was easy to say that the data frame represents a time-series in case there were two values in a row: a timestamp and a value. However with the introduction of the `wide` representation (multiple metrics for a single timestamp) this heuristic doesn't work anymore.

Reply:
Resolving this one as this part got removed from the guide. ([More info and explanation](https://github.com/grafana/grafana/pull/38911#discussion_r708992021))